fed/mosaic-application/src/main/java/org/eclipse/mosaic/fed/application/ambassador/ApplicationAmbassador.java

```diff
@@ -449,6 +460,8 @@ private void process(final ChargingStationUpdate chargingStationUpdate) {
         final AbstractSimulationUnit simulationUnit =
                 UnitSimulator.UnitSimulator.getUnitFromId(chargingStationData.getName());
 
+        SimulationKernel.SimulationKernel.chargingStationIndex.updateChargingStation(chargingStationUpdate.getUpdatedChargingStation());
```

Comment:
Use `SimulationKernel.SimulationKernel.getChargingStationIndex()` here.
Also why don't you just use the `chargingStationData` as input declared in line 459?